File: backend/app/dtos/recipe_read.py
from pydantic import BaseModel, ConfigDict
from typing import List
from .recipe_ingredient_read import RecipeIngredientRead

class RecipeRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    profit_per_craft: int = 0
    profession_id: int
    expansion_id: int
    ingredients: List[RecipeIngredientRead] = []

File: frontend/src/dto/RecipeRead.ts
import type { IngredientRead } from "./IngredientRead";

export interface RecipeRead {
    id: number;
    name: string;
    profit_per_craft: number;
    profession_id: number;
    expansion_id: number;
    ingredients: RecipeIngredientRead[];
}

export interface RecipeIngredientRead {
    amount_required: number;
    ingredient: IngredientRead;
}
